### PR TITLE
fix projects with empty spaces aspire

### PR DESCRIPTION
**Pull request by vhvb1989**
Re-use the same logic to name a project from easy-init for the AppHost's project name generation
fix: https://github.com/Azure/azure-dev/issues/4470

### CODE REVIEW

File: cli/azd/pkg/apphost/generate.go [hunk 1]
```diff
@@ -439,7 +440,7 @@ func GenerateProjectArtifacts(
 	generatedFS := memfs.New()
 
 	projectFileContext := genProjectFileContext{
-		Name: projectName,
+		Name: label.LabelName(projectName),
```
Comment (weikanglim) [inline cli/azd/pkg/apphost/generate.go:443]: I think I'd prefer to do this at the layer where we pass in the name (see list of files below). My inclination here is that name sanitization/restriction enforcement is really important to be done at closest to the user interaction layer as possible.

- [environment_service_create.go](https://github.com/Azure/azure-dev/blob/4182b4138151b8fac4e9d5e6f6220b32dddace80/cli/azd/internal/vsrpc/environment_service_create.go#L73)
- [app_init.go](https://github.com/Azure/azure-dev/blob/4182b4138151b8fac4e9d5e6f6220b32dddace80/cli/azd/internal/repository/app_init.go#L176).

Comment (vhvb1989) [reply cli/azd/pkg/apphost/generate.go]: moved the logic to `GetDefaultProjectName()`

File: cli/azd/internal/vsrpc/environment_service_create.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func (s *environmentService) CreateEnvironmentAsync(
 			return false, fmt.Errorf("reading app host manifest: %w", err)
 		}
 
-		projectName := strings.TrimSuffix(filepath.Base(c.azdContext.ProjectDirectory()), ".AppHost")
+		projectName := strings.TrimSuffix(c.azdContext.GetDefaultProjectName(), ".AppHost")
```
Comment (weikanglim) [inline cli/azd/internal/vsrpc/environment_service_create.go:70]: Just wondering, would this code still make sense the refactor to `GetDefaultProjectName`? `GetDefaultProjectName` is now an implementation detail that we then use to trim the`.AppHost` suffix?

File: cli/azd/pkg/apphost/generate.go [hunk 3]
```diff
@@ -431,6 +431,13 @@ func GenerateProjectArtifacts(
 	manifest *Manifest,
 	appHostProject string,
 ) (map[string]ContentsAndMode, error) {
+	// panic if projectName contains spaces. This is because the project name is used to generate the name of containers
+	// and other resources, and spaces are not allowed in those names.
+	if strings.Contains(projectName, " ") {
```
Comment (weikanglim) [inline cli/azd/pkg/apphost/generate.go:436]: Just wondering, how would we maintain this code over time?

There are many, many other disallowed characters. `apphost.generate` knows not a lot about project name validation.

I'm also fine if we want to track the removal with the linked naming issue #4400. I would lean towards just removing this -- we can catch this via code review right now.

File: cli/azd/internal/vsrpc/environment_service_create.go (hunk 2 above)
Comment (weikanglim) [reply cli/azd/internal/vsrpc/environment_service_create.go:70]: The current change actually regresses functionality -- we would have replaced `.` with `-` and `TrimSuffix("my-app-host", ".AppHost")` would not trim the actual suffix.

Comment (weikanglim) [reply cli/azd/internal/vsrpc/environment_service_create.go:70]: My coded up suggestions here:

https://github.com/vhvb1989/azure-dev/compare/fix-projects-with-empty-spaces-aspire...weikanglim:weilim/aspire-naming?expand=1

**Review by weikanglim — APPROVED** [review]
LGTM

**Pull request merged**